File: qc-core/src/main/java/com/eager/questioncloud/core/domain/subscribe/repository/SubscribeRepository.java
package com.eager.questioncloud.core.domain.subscribe.repository;

import com.eager.questioncloud.core.common.PagingInformation;
import com.eager.questioncloud.core.domain.subscribe.dto.SubscribeDto.SubscribeListItem;
import com.eager.questioncloud.core.domain.subscribe.model.Subscribe;
import java.util.List;

public interface SubscribeRepository {
    Subscribe save(Subscribe subscribe);

    Boolean isSubscribed(Long subscriberId, Long creatorId);

    void unSubscribe(Long subscriberId, Long creatorId);

    int countSubscriber(Long creatorId);

    List<SubscribeListItem> getSubscribeCreators(Long userId, PagingInformation pagingInformation);

    int countMySubscribe(Long userId);
}

File: qc-core/qc-subscribe/qc-subscribe-core/src/main/kotlin/com/eager/questioncloud/subscribe/infrastructure/repository/SubscribeRepository.kt
package com.eager.questioncloud.subscribe.infrastructure.repository

import com.eager.questioncloud.common.pagination.PagingInformation
import com.eager.questioncloud.subscribe.domain.Subscribe

interface SubscribeRepository {
    fun save(subscribe: Subscribe): Subscribe

    fun isSubscribed(subscriberId: Long, creatorId: Long): Boolean

    fun unSubscribe(subscriberId: Long, creatorId: Long)

    fun countSubscriber(creatorId: Long): Int

    fun countSubscriber(creatorIds: List<Long>): Map<Long, Int>

    fun getMySubscribedCreators(userId: Long, pagingInformation: PagingInformation): List<Long>

    fun countMySubscribe(userId: Long): Int

    fun deleteAllInBatch()
}
